# Introduction

New York City is one of the largest cities in the world with 8 million people. There are five boroughs: Manhattan, Brooklyn, Queens, Bronx, and Staten Island. As a resident in NYC, I decided to explore this city in my project. One interesting idea would be, if one wants to purchase an apartment in New York City, what are the best neighborhood to consider? The tradeoff is usually between price and convenience. The more centrally located neighborhoods with a lot of restaurants and grocery stores are often more expensive, while more affordable neighborhoods tend to be further from the city center, less convenient with less shops and restaurants. 

With all the above in mind, I will create a ranking of neighborhoods according to their housing prices, distance to city center, number of shops, restaurants, and grocery stores. For simplicity, I will only consider the neighborhoods in Manhattan.

# Data

The data to explore this question come from below sources:

**1.Housing price data**

Housing price data is downloaded from StreetEasy Data Dashboard: https://streeteasy.com/blog/data-dashboard/. It provides downloadable csv files for price index, median sale price, median asking price for each neighborhood. Since median sale price data has a lot of missing values if there is no sales in a given month, I will just use the median asking price as a gauge of housing price. The data is a time-series data of monthly prices and I just use the latest available month, which is December 2019.

**2.Venue data**

I use Foursquare API to get venue data such as the number of shops, restaurants and grocery stores in each neighborhood.

**3.Distance data**

I use Google map to get distance to city center data for each neighborhood. The official New York City coordinates are those of the city hall, but most professional people work around midtown, and when we consider distance to center we are mostly concerned about commute for work, so I use midtown Manhattan coordinates 40.7549° N, 73.9840° W to calculate distance to center.

# Methodology

## Import Data

I download the Manhattan median asking price by neighborhood data from StreetEasy and combine it with distance to center data from Google Map in one csv file:

In [138]:
import pandas as pd
import numpy as np

In [139]:
df=pd.read_csv('neighborhoods.csv')
df

Neighborhood      Price  DistanceToCenter
0     Battery Park City  1340000.0               3.8
1        Central Harlem   980000.0               4.5
2    Central Park South  2345000.0               0.9
3               Chelsea  2275000.0               1.4
4             Chinatown  1995000.0               3.0
5          Civic Center        NaN               3.1
6           East Harlem   950000.0               4.3
7          East Village  1250000.0               2.4
8    Financial District  1397000.0               3.7
9              Flatiron  2695000.0               1.1
10        Gramercy Park  1715000.0               1.5
11    Greenwich Village  1975000.0               1.8
12     Hamilton Heights   780000.0               5.6
13               Inwood   487500.0               8.9
14         Little Italy        NaN               2.7
15      Lower East Side  1397500.0               3.3
16          Marble Hill        NaN               9.9
17              Midtown  2225000.0               0.0
18         Midtown East   950000.0               0.9
19        Midtown South  1565550.0               0.3
20         Midtown West  1492500.0               0.9
21  Morningside Heights   799000.0               4.3
22               Nolita  4750000.0               2.5
23     Roosevelt Island  1089000.0               3.4
24                 Soho  3495000.0               2.5
25  Stuyvesant Town/PCV        NaN               2.1
26              Tribeca  4274500.0               3.0
27      Upper East Side  1825000.0               2.5
28      Upper West Side  1972500.0               2.5
29   Washington Heights   547000.0               6.8
30          West Harlem  1155000.0               5.1
31         West Village  2100000.0               1.9

We can see there are 32 neighborhoods in Manhattan. Let''s drop neighborhoods that have missing values:

In [140]:
df.dropna(inplace=True)
df

Neighborhood      Price  DistanceToCenter
0     Battery Park City  1340000.0               3.8
1        Central Harlem   980000.0               4.5
2    Central Park South  2345000.0               0.9
3               Chelsea  2275000.0               1.4
4             Chinatown  1995000.0               3.0
6           East Harlem   950000.0               4.3
7          East Village  1250000.0               2.4
8    Financial District  1397000.0               3.7
9              Flatiron  2695000.0               1.1
10        Gramercy Park  1715000.0               1.5
11    Greenwich Village  1975000.0               1.8
12     Hamilton Heights   780000.0               5.6
13               Inwood   487500.0               8.9
15      Lower East Side  1397500.0               3.3
17              Midtown  2225000.0               0.0
18         Midtown East   950000.0               0.9
19        Midtown South  1565550.0               0.3
20         Midtown West  1492500.0               0.9
21  Morningside Heights   799000.0               4.3
22               Nolita  4750000.0               2.5
23     Roosevelt Island  1089000.0               3.4
24                 Soho  3495000.0               2.5
26              Tribeca  4274500.0               3.0
27      Upper East Side  1825000.0               2.5
28      Upper West Side  1972500.0               2.5
29   Washington Heights   547000.0               6.8
30          West Harlem  1155000.0               5.1
31         West Village  2100000.0               1.9

## Get the latitude and longitude coordinates of each neighborhood

In [141]:
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

In [142]:
neighborhoods=df['Neighborhood'].tolist()
geolocator = Nominatim(user_agent="newyork_explorer")
latitude=[]
longitude=[]
for n in neighborhoods:
    address = n+', New York'
    location = geolocator.geocode(address)
    latitude.append(location.latitude)
    longitude.append(location.longitude)
df['Latitude']=latitude
df['Longitude']=longitude
df

Neighborhood      Price  DistanceToCenter   Latitude  Longitude
0     Battery Park City  1340000.0               3.8  40.711017 -74.016937
1        Central Harlem   980000.0               4.5  40.807879 -73.945415
2    Central Park South  2345000.0               0.9  40.767750 -73.981318
3               Chelsea  2275000.0               1.4  40.746491 -74.001528
4             Chinatown  1995000.0               3.0  40.716491 -73.996250
6           East Harlem   950000.0               4.3  40.794722 -73.942500
7          East Village  1250000.0               2.4  40.729269 -73.987361
8    Financial District  1397000.0               3.7  40.707612 -74.009378
9              Flatiron  2695000.0               1.1  40.741059 -73.989642
10        Gramercy Park  1715000.0               1.5  40.737925 -73.985932
11    Greenwich Village  1975000.0               1.8  40.731980 -73.996566
12     Hamilton Heights   780000.0               5.6  40.824145 -73.950062
13               Inwood   487500.0               8.9  40.869258 -73.920495
15      Lower East Side  1397500.0               3.3  40.715936 -73.986806
17              Midtown  2225000.0               0.0  40.762268 -73.979544
18         Midtown East   950000.0               0.9  40.759822 -73.972471
19        Midtown South  1565550.0               0.3  40.749842 -73.984251
20         Midtown West  1492500.0               0.9  40.764423 -73.992392
21  Morningside Heights   799000.0               4.3  40.810000 -73.962500
22               Nolita  4750000.0               2.5  40.720370 -73.994582
23     Roosevelt Island  1089000.0               3.4  40.761418 -73.950228
24                 Soho  3495000.0               2.5  40.722880 -73.998750
26              Tribeca  4274500.0               3.0  40.715380 -74.009306
27      Upper East Side  1825000.0               2.5  40.773702 -73.964120
28      Upper West Side  1972500.0               2.5  40.787045 -73.975416
29   Washington Heights   547000.0               6.8  40.840198 -73.940221
30          West Harlem  1155000.0               5.1  40.807879 -73.945415
31         West Village  2100000.0               1.9  40.734186 -74.005580

## Create a map of Manhattan New York with neighborhoods superimposed on top

In [143]:
import folium
# create map of New York using latitude and longitude values
map_newyork = folium.Map(location=[40.762268,-73.979544], zoom_start=12)

# add markers to map
for lat, lng, neighborhood in zip(df['Latitude'], df['Longitude'], df['Neighborhood']):
    label = '{}'.format(neighborhood)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_newyork)  
    
map_newyork

## Foursquare

Now we can find venues in each neighborhood using Foursquare API. First let's define Foursquare credentials and Version:

In [144]:
CLIENT_ID = 'OKIBPTJLJIFBJDZR3NN0CYN0YWG35ANBIEGDBQWDOQCWN10G' # your Foursquare ID
CLIENT_SECRET = 'NC4WTAGQX23WZ1UYSK5YYDEHVV1L2PKXDKFTK4VYWOZRT5MV' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: OKIBPTJLJIFBJDZR3NN0CYN0YWG35ANBIEGDBQWDOQCWN10G
CLIENT_SECRET:NC4WTAGQX23WZ1UYSK5YYDEHVV1L2PKXDKFTK4VYWOZRT5MV


In [145]:
# Create a function to explore venues
def getNearbyVenues(names, latitudes, longitudes, category, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&categoryId={}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            category,
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

## Find number of restaurants in each neighborhood

In [146]:
import requests
# run the above function on each neighborhood and create a new dataframe 
LIMIT = 200 # limit of number of venues returned by Foursquare API
radius = 500 # define radius
food_category='4d4b7105d754a06374d81259' # 'Root' category for all food-related venues
nyc_restaurants = getNearbyVenues(names=df['Neighborhood'],
                                   latitudes=df['Latitude'],
                                   longitudes=df['Longitude'],
                                   category=food_category
                                  )
nyc_restaurants.shape

Battery Park City
Central Harlem
Central Park South
Chelsea
Chinatown
East Harlem
East Village
Financial District
Flatiron
Gramercy Park
Greenwich Village
Hamilton Heights
Inwood
Lower East Side
Midtown
Midtown East
Midtown South
Midtown West
Morningside Heights
Nolita
Roosevelt Island
Soho
Tribeca
Upper East Side
Upper West Side
Washington Heights
West Harlem
West Village


(2233, 7)

There are a total of 2233 restaurants in Manhattan. Now let''s calculate the number of restaurants in each neighborhood.

In [147]:
manhattan=pd.merge(df,nyc_restaurants.groupby('Neighborhood',as_index=False).count()[['Neighborhood','Venue']])
manhattan.rename(columns={'Venue':'NumRestaurant'},inplace=True)
manhattan

Neighborhood      Price  DistanceToCenter   Latitude  Longitude  \
0     Battery Park City  1340000.0               3.8  40.711017 -74.016937   
1        Central Harlem   980000.0               4.5  40.807879 -73.945415   
2    Central Park South  2345000.0               0.9  40.767750 -73.981318   
3               Chelsea  2275000.0               1.4  40.746491 -74.001528   
4             Chinatown  1995000.0               3.0  40.716491 -73.996250   
5           East Harlem   950000.0               4.3  40.794722 -73.942500   
6          East Village  1250000.0               2.4  40.729269 -73.987361   
7    Financial District  1397000.0               3.7  40.707612 -74.009378   
8              Flatiron  2695000.0               1.1  40.741059 -73.989642   
9         Gramercy Park  1715000.0               1.5  40.737925 -73.985932   
10    Greenwich Village  1975000.0               1.8  40.731980 -73.996566   
11     Hamilton Heights   780000.0               5.6  40.824145 -73.950062   
12               Inwood   487500.0               8.9  40.869258 -73.920495   
13      Lower East Side  1397500.0               3.3  40.715936 -73.986806   
14              Midtown  2225000.0               0.0  40.762268 -73.979544   
15         Midtown East   950000.0               0.9  40.759822 -73.972471   
16        Midtown South  1565550.0               0.3  40.749842 -73.984251   
17         Midtown West  1492500.0               0.9  40.764423 -73.992392   
18  Morningside Heights   799000.0               4.3  40.810000 -73.962500   
19               Nolita  4750000.0               2.5  40.720370 -73.994582   
20     Roosevelt Island  1089000.0               3.4  40.761418 -73.950228   
21                 Soho  3495000.0               2.5  40.722880 -73.998750   
22              Tribeca  4274500.0               3.0  40.715380 -74.009306   
23      Upper East Side  1825000.0               2.5  40.773702 -73.964120   
24      Upper West Side  1972500.0               2.5  40.787045 -73.975416   
25   Washington Heights   547000.0               6.8  40.840198 -73.940221   
26          West Harlem  1155000.0               5.1  40.807879 -73.945415   
27         West Village  2100000.0               1.9  40.734186 -74.005580   

    NumRestaurant  
0              70  
1              68  
2             100  
3              66  
4             100  
5              49  
6             100  
7             100  
8             100  
9             100  
10            100  
11             64  
12             41  
13            100  
14            100  
15            100  
16            100  
17            100  
18             38  
19            100  
20             14  
21            100  
22             88  
23             54  
24             64  
25             49  
26             68  
27            100

## Find number of grocery stores in each neighborhood

In [148]:
grocery_category='4bf58dd8d48988d118951735' # category for grocery stores from https://developer.foursquare.com/docs/resources/categories
nyc_grocery = getNearbyVenues(names=df['Neighborhood'],
                                   latitudes=df['Latitude'],
                                   longitudes=df['Longitude'],
                                   category=grocery_category
                                  )
nyc_grocery.shape

Battery Park City
Central Harlem
Central Park South
Chelsea
Chinatown
East Harlem
East Village
Financial District
Flatiron
Gramercy Park
Greenwich Village
Hamilton Heights
Inwood
Lower East Side
Midtown
Midtown East
Midtown South
Midtown West
Morningside Heights
Nolita
Roosevelt Island
Soho
Tribeca
Upper East Side
Upper West Side
Washington Heights
West Harlem
West Village


(310, 7)

There are a total of 310 grocery stores in Manhattan. Now let''s calculate the number of grocery stores in each neighborhood.

In [149]:
manhattan2=pd.merge(manhattan,nyc_grocery.groupby('Neighborhood',as_index=False).count()[['Neighborhood','Venue']])
manhattan2.rename(columns={'Venue':'NumGrocery'},inplace=True)
manhattan2

Neighborhood      Price  DistanceToCenter   Latitude  Longitude  \
0     Battery Park City  1340000.0               3.8  40.711017 -74.016937   
1        Central Harlem   980000.0               4.5  40.807879 -73.945415   
2    Central Park South  2345000.0               0.9  40.767750 -73.981318   
3               Chelsea  2275000.0               1.4  40.746491 -74.001528   
4             Chinatown  1995000.0               3.0  40.716491 -73.996250   
5           East Harlem   950000.0               4.3  40.794722 -73.942500   
6          East Village  1250000.0               2.4  40.729269 -73.987361   
7    Financial District  1397000.0               3.7  40.707612 -74.009378   
8              Flatiron  2695000.0               1.1  40.741059 -73.989642   
9         Gramercy Park  1715000.0               1.5  40.737925 -73.985932   
10    Greenwich Village  1975000.0               1.8  40.731980 -73.996566   
11     Hamilton Heights   780000.0               5.6  40.824145 -73.950062   
12               Inwood   487500.0               8.9  40.869258 -73.920495   
13      Lower East Side  1397500.0               3.3  40.715936 -73.986806   
14              Midtown  2225000.0               0.0  40.762268 -73.979544   
15         Midtown East   950000.0               0.9  40.759822 -73.972471   
16        Midtown South  1565550.0               0.3  40.749842 -73.984251   
17         Midtown West  1492500.0               0.9  40.764423 -73.992392   
18  Morningside Heights   799000.0               4.3  40.810000 -73.962500   
19               Nolita  4750000.0               2.5  40.720370 -73.994582   
20     Roosevelt Island  1089000.0               3.4  40.761418 -73.950228   
21                 Soho  3495000.0               2.5  40.722880 -73.998750   
22              Tribeca  4274500.0               3.0  40.715380 -74.009306   
23      Upper East Side  1825000.0               2.5  40.773702 -73.964120   
24      Upper West Side  1972500.0               2.5  40.787045 -73.975416   
25   Washington Heights   547000.0               6.8  40.840198 -73.940221   
26          West Harlem  1155000.0               5.1  40.807879 -73.945415   
27         West Village  2100000.0               1.9  40.734186 -74.005580   

    NumRestaurant  NumGrocery  
0              70           7  
1              68           5  
2             100           7  
3              66          10  
4             100          65  
5              49          14  
6             100          15  
7             100          12  
8             100          11  
9             100          10  
10            100           9  
11             64           1  
12             41           3  
13            100          16  
14            100           6  
15            100           3  
16            100           6  
17            100           7  
18             38           3  
19            100          44  
20             14           1  
21            100          13  
22             88           6  
23             54           5  
24             64          11  
25             49           8  
26             68           5  
27            100           7

In [150]:
manhattan2.describe()

Price  DistanceToCenter   Latitude  Longitude  NumRestaurant  \
count  2.800000e+01         28.000000  28.000000  28.000000      28.000000   
mean   1.779359e+06          2.957143  40.761105 -73.978035      79.750000   
std    1.031887e+06          2.012264   0.042684   0.025049      25.383904   
min    4.875000e+05          0.000000  40.707612 -74.016937      14.000000   
25%    1.061750e+06          1.475000  40.727672 -73.996329      64.000000   
50%    1.529025e+06          2.500000  40.754832 -73.985092     100.000000   
75%    2.131250e+06          3.925000  40.788965 -73.959432     100.000000   
max    4.750000e+06          8.900000  40.869258 -73.920495     100.000000   

       NumGrocery  
count   28.000000  
mean    11.071429  
std     13.213149  
min      1.000000  
25%      5.000000  
50%      7.000000  
75%     11.250000  
max     65.000000

## Rank neighborhoods

Now we have the complete data for each neighborhood, including housing price, distance to center, number of restaurants and number of grocery. Next we will rank the neighborhoods by these four metrics. I do this by first ranking each neighborhood according to each individual metric. Then assuming equal weight for each metric, I will sum up all four ranks to get an aggregate score. The smaller the aggregate score, the higher the rank of a particular neighborhood.

In [151]:
#Create rank columns for each metric
manhattan2['RankPrice']=manhattan2['Price'].rank()
manhattan2['RankDistance']=manhattan2['DistanceToCenter'].rank()
manhattan2['RankRes']=manhattan2['NumRestaurant'].rank(ascending=False)
manhattan2['RankGrocery']=manhattan2['NumGrocery'].rank(ascending=False)
manhattan2

Neighborhood      Price  DistanceToCenter   Latitude  Longitude  \
0     Battery Park City  1340000.0               3.8  40.711017 -74.016937   
1        Central Harlem   980000.0               4.5  40.807879 -73.945415   
2    Central Park South  2345000.0               0.9  40.767750 -73.981318   
3               Chelsea  2275000.0               1.4  40.746491 -74.001528   
4             Chinatown  1995000.0               3.0  40.716491 -73.996250   
5           East Harlem   950000.0               4.3  40.794722 -73.942500   
6          East Village  1250000.0               2.4  40.729269 -73.987361   
7    Financial District  1397000.0               3.7  40.707612 -74.009378   
8              Flatiron  2695000.0               1.1  40.741059 -73.989642   
9         Gramercy Park  1715000.0               1.5  40.737925 -73.985932   
10    Greenwich Village  1975000.0               1.8  40.731980 -73.996566   
11     Hamilton Heights   780000.0               5.6  40.824145 -73.950062   
12               Inwood   487500.0               8.9  40.869258 -73.920495   
13      Lower East Side  1397500.0               3.3  40.715936 -73.986806   
14              Midtown  2225000.0               0.0  40.762268 -73.979544   
15         Midtown East   950000.0               0.9  40.759822 -73.972471   
16        Midtown South  1565550.0               0.3  40.749842 -73.984251   
17         Midtown West  1492500.0               0.9  40.764423 -73.992392   
18  Morningside Heights   799000.0               4.3  40.810000 -73.962500   
19               Nolita  4750000.0               2.5  40.720370 -73.994582   
20     Roosevelt Island  1089000.0               3.4  40.761418 -73.950228   
21                 Soho  3495000.0               2.5  40.722880 -73.998750   
22              Tribeca  4274500.0               3.0  40.715380 -74.009306   
23      Upper East Side  1825000.0               2.5  40.773702 -73.964120   
24      Upper West Side  1972500.0               2.5  40.787045 -73.975416   
25   Washington Heights   547000.0               6.8  40.840198 -73.940221   
26          West Harlem  1155000.0               5.1  40.807879 -73.945415   
27         West Village  2100000.0               1.9  40.734186 -74.005580   

    NumRestaurant  NumGrocery  RankPrice  RankDistance  RankRes  RankGrocery  
0              70           7       11.0          21.0     17.0         15.5  
1              68           5        7.0          24.0     18.5         22.0  
2             100           7       24.0           4.0      8.0         15.5  
3              66          10       23.0           7.0     20.0         10.5  
4             100          65       20.0          16.5      8.0          1.0  
5              49          14        5.5          22.5     24.5          5.0  
6             100          15       10.0          11.0      8.0          4.0  
7             100          12       12.0          20.0      8.0          7.0  
8             100          11       25.0           6.0      8.0          8.5  
9             100          10       16.0           8.0      8.0         10.5  
10            100           9       19.0           9.0      8.0         12.0  
11             64           1        3.0          26.0     21.5         27.5  
12             41           3        1.0          28.0     26.0         25.0  
13            100          16       13.0          18.0      8.0          3.0  
14            100           6       22.0           1.0      8.0         19.0  
15            100           3        5.5           4.0      8.0         25.0  
16            100           6       15.0           2.0      8.0         19.0  
17            100           7       14.0           4.0      8.0         15.5  
18             38           3        4.0          22.5     27.0         25.0  
19            100          44       28.0          13.5      8.0          2.0  
20             14           1        8.0          19.0     28.0         27.5  
21        

In [152]:
#Create overall rank by adding all four ranks
manhattan2['OverallRank']=manhattan2[['RankPrice','RankDistance','RankRes','RankGrocery']].sum(axis=1)
manhattan2

Neighborhood      Price  DistanceToCenter   Latitude  Longitude  \
0     Battery Park City  1340000.0               3.8  40.711017 -74.016937   
1        Central Harlem   980000.0               4.5  40.807879 -73.945415   
2    Central Park South  2345000.0               0.9  40.767750 -73.981318   
3               Chelsea  2275000.0               1.4  40.746491 -74.001528   
4             Chinatown  1995000.0               3.0  40.716491 -73.996250   
5           East Harlem   950000.0               4.3  40.794722 -73.942500   
6          East Village  1250000.0               2.4  40.729269 -73.987361   
7    Financial District  1397000.0               3.7  40.707612 -74.009378   
8              Flatiron  2695000.0               1.1  40.741059 -73.989642   
9         Gramercy Park  1715000.0               1.5  40.737925 -73.985932   
10    Greenwich Village  1975000.0               1.8  40.731980 -73.996566   
11     Hamilton Heights   780000.0               5.6  40.824145 -73.950062   
12               Inwood   487500.0               8.9  40.869258 -73.920495   
13      Lower East Side  1397500.0               3.3  40.715936 -73.986806   
14              Midtown  2225000.0               0.0  40.762268 -73.979544   
15         Midtown East   950000.0               0.9  40.759822 -73.972471   
16        Midtown South  1565550.0               0.3  40.749842 -73.984251   
17         Midtown West  1492500.0               0.9  40.764423 -73.992392   
18  Morningside Heights   799000.0               4.3  40.810000 -73.962500   
19               Nolita  4750000.0               2.5  40.720370 -73.994582   
20     Roosevelt Island  1089000.0               3.4  40.761418 -73.950228   
21                 Soho  3495000.0               2.5  40.722880 -73.998750   
22              Tribeca  4274500.0               3.0  40.715380 -74.009306   
23      Upper East Side  1825000.0               2.5  40.773702 -73.964120   
24      Upper West Side  1972500.0               2.5  40.787045 -73.975416   
25   Washington Heights   547000.0               6.8  40.840198 -73.940221   
26          West Harlem  1155000.0               5.1  40.807879 -73.945415   
27         West Village  2100000.0               1.9  40.734186 -74.005580   

    NumRestaurant  NumGrocery  RankPrice  RankDistance  RankRes  RankGrocery  \
0              70           7       11.0          21.0     17.0         15.5   
1              68           5        7.0          24.0     18.5         22.0   
2             100           7       24.0           4.0      8.0         15.5   
3              66          10       23.0           7.0     20.0         10.5   
4             100          65       20.0          16.5      8.0          1.0   
5              49          14        5.5          22.5     24.5          5.0   
6             100          15       10.0          11.0      8.0          4.0   
7             100          12       12.0          20.0      8.0          7.0   
8             100          11       25.0           6.0      8.0          8.5   
9             100          10       16.0           8.0      8.0         10.5   
10            100           9       19.0           9.0      8.0         12.0   
11             64           1        3.0          26.0     21.5         27.5   
12             41           3        1.0          28.0     26.0         25.0   
13            100          16       13.0          18.0      8.0          3.0   
14            100           6       22.0           1.0      8.0         19.0   
15            100           3        5.5           4.0      8.0         25.0   
16            100           6       15.0           2.0      8.0         19.0   
17            100           7       14.0           4.0      8.0         15.5   
18             38           3        4.0          22.5     27.0         25.0   
19            100          44       28.0          13.5      8.0          2.0   
20             14           1        8.0          19.0     28.0     

By sorting the overall rank by ascending order, we can get the rank of the neighborhoods:

In [153]:
#sort by overall rank
manhattan2.sort_values(by=['OverallRank'],inplace=True)
manhattan2

Neighborhood      Price  DistanceToCenter   Latitude  Longitude  \
6          East Village  1250000.0               2.4  40.729269 -73.987361   
17         Midtown West  1492500.0               0.9  40.764423 -73.992392   
13      Lower East Side  1397500.0               3.3  40.715936 -73.986806   
9         Gramercy Park  1715000.0               1.5  40.737925 -73.985932   
15         Midtown East   950000.0               0.9  40.759822 -73.972471   
16        Midtown South  1565550.0               0.3  40.749842 -73.984251   
4             Chinatown  1995000.0               3.0  40.716491 -73.996250   
7    Financial District  1397000.0               3.7  40.707612 -74.009378   
8              Flatiron  2695000.0               1.1  40.741059 -73.989642   
10    Greenwich Village  1975000.0               1.8  40.731980 -73.996566   
14              Midtown  2225000.0               0.0  40.762268 -73.979544   
2    Central Park South  2345000.0               0.9  40.767750 -73.981318   
19               Nolita  4750000.0               2.5  40.720370 -73.994582   
21                 Soho  3495000.0               2.5  40.722880 -73.998750   
27         West Village  2100000.0               1.9  40.734186 -74.005580   
5           East Harlem   950000.0               4.3  40.794722 -73.942500   
3               Chelsea  2275000.0               1.4  40.746491 -74.001528   
24      Upper West Side  1972500.0               2.5  40.787045 -73.975416   
0     Battery Park City  1340000.0               3.8  40.711017 -74.016937   
25   Washington Heights   547000.0               6.8  40.840198 -73.940221   
1        Central Harlem   980000.0               4.5  40.807879 -73.945415   
26          West Harlem  1155000.0               5.1  40.807879 -73.945415   
23      Upper East Side  1825000.0               2.5  40.773702 -73.964120   
11     Hamilton Heights   780000.0               5.6  40.824145 -73.950062   
18  Morningside Heights   799000.0               4.3  40.810000 -73.962500   
22              Tribeca  4274500.0               3.0  40.715380 -74.009306   
12               Inwood   487500.0               8.9  40.869258 -73.920495   
20     Roosevelt Island  1089000.0               3.4  40.761418 -73.950228   

    NumRestaurant  NumGrocery  RankPrice  RankDistance  RankRes  RankGrocery  \
6             100          15       10.0          11.0      8.0          4.0   
17            100           7       14.0           4.0      8.0         15.5   
13            100          16       13.0          18.0      8.0          3.0   
9             100          10       16.0           8.0      8.0         10.5   
15            100           3        5.5           4.0      8.0         25.0   
16            100           6       15.0           2.0      8.0         19.0   
4             100          65       20.0          16.5      8.0          1.0   
7             100          12       12.0          20.0      8.0          7.0   
8             100          11       25.0           6.0      8.0          8.5   
10            100           9       19.0           9.0      8.0         12.0   
14            100           6       22.0           1.0      8.0         19.0   
2             100           7       24.0           4.0      8.0         15.5   
19            100          44       28.0          13.5      8.0          2.0   
21            100          13       26.0          13.5      8.0          6.0   
27            100           7       21.0          10.0      8.0         15.5   
5              49          14        5.5          22.5     24.5          5.0   
3              66          10       23.0           7.0     20.0         10.5   
24             64          11       18.0          13.5     21.5          8.5   
0              70           7       11.0          21.0     17.0         15.5   
25             49           8        2.0          27.0     24.5         13.0   
1              68           5        7.0          24.0     18.5     

Now we see that the top 5 neighborhoods to consider to purchase an apartment are: East Village, Midtown West, Lower East Side, Gramercy Park, and Midtown East. We can visualize these top choices on the map:

In [154]:
# add markers to map
# create map of New York using latitude and longitude values
map_manhattan = folium.Map(location=[40.762268,-73.979544], zoom_start=12)

for lat, lng in zip(manhattan2['Latitude'].head(5), manhattan2['Longitude'].head(5)):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color='red',
        fill=True,
        fill_color='red',
        fill_opacity=0.5,
        parse_html=False).add_to(map_manhattan)  

map_manhattan

# Results and Discussion

In the above analysis, we collect and create data for each of the 32 neighborhoods in Manhattan, New York their median housing price, distance to city center, number of restaurants and number of grocery stores. We then rank the neighborhoods by each of the four metrics that we think are most relevant in making decision on purchasing an apartment. Finally we combine the four ranks by equal weight to get an overall rank. We found that the top 5 neighborhoods in Manhattan are East Village, Midtown West, Lower East Side, Gramercy Park, and Midtown East. These candidates overall represent the neighborhoods that are most affordable, centrally located, and abundant in restaurants and grocery stores.

Of course, we can add more metrics to evaluate the neighborhoods, depending on the target client's preferences. For example, we can add the number of subway stations nearby to represent the convenience of commute, number of movie theaters for movie goers, number of gyms for athletic people, number of bars to represent the abundance of nightlife, or school zone rating for families with school-aged children.

In addition, we can also apply different weightings for the selected metrics. One can easily adjust up or down the individual weighting of each metric according to their own preferences. For example, if the target client is more sensitive to housing price, then we can apply a higher weight on the price metric etc. The model is very flexible.

# Conclusion

In this project, we try to answer the question: what's the best neighborhood to consider if we want to purchase an apartment in Manhattan? We present a simple approach by collecting data of four different metrics: housing price, distance to city center, number of restaurants, and number of grocery stores. By leveraging Foursquare API data, we are able to calculate the number of restaurants and grocery stores in each neighborhood. We then rank the neighborhood by each metric and sum them together to get the overall score. We find that the best neighborhood options are East Village, Midtown West, Lower East Side, Gramercy Park, and Midtown East.

The model is flexible and can be easily modified according the stakeholder's preferences. Using the same framework, we can collect data on different metrics and apply different weights on each metric. We can also cluster the neighborhoods to find commonalities among the neighborhoods as an extension of the project.